crates/prover/src/runner.rs

```diff
@@ -492,14 +491,25 @@ where
                 break;
             }
 
+            // if proof_sampling_number is 0, then we always prove and submit
+            // otherwise we submit and prove with a probability of 1/proof_sampling_number
+            let should_prove = prover_config.proof_sampling_number == 0
+                || rand::thread_rng().gen_range(0..prover_config.proof_sampling_number) == 0;
+
+            let mut sequencer_commitments =
+                self.filter_out_proven_commitments(&sequencer_commitments)?;
+            // Make sure all sequencer commitments are stored in ascending order.
+            sequencer_commitments.sort_unstable();
```

Comment:
But we still can have duplicates of unproved commitments.

Reply:
The line you're addressing is irrelevant & you don't provide details on why you think that is the case. Can you elaborate please?

Reply:
No idea where to put my comment. So I left it there.

1. sequencer pushes a sequencer commitment on DA
2. mallory scans mempool and pushes a copy of sequencer commitment on DA

Reply:
Ok so AFAIU 2 transactions with the same sequencer commitment into the DA layer. What happens is that one or both of these transactions is going to be included into the DA layer blocks. Here we have 2 cases:

Case 1: Commitment txs are included in different DA blocks:
In this case, the fullnode & prover will read the first block,extract commitments and mark L2 range as finalized. The second block containing the replayed commitment would be filtered out since the L2 range is already finalized.

We agree this is handled?

Case 2: Commitment txs are included in the same DA block

What we can do in this case is to filter out commitments which have the same L2 range that we've just extracted from the current DA block. what do you think?

Reply:
1) yes
2) yes. I believe we don't care which one seq commitment to prove as long as they are valid

Reply:
Handled that scenario